src/Test/Diagnostics/Waiters/EditorAdornmentWaiter.cs

```diff
@@ -1,26 +0,0 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
-using System;
-using System.Threading;
-using System.Threading.Tasks;
-using System.Windows.Threading;
-using Microsoft.CodeAnalysis.Shared.TestHooks;
-using Roslyn.Utilities;
-
-namespace Roslyn.Hosting.Diagnostics.Waiters
-{
-    internal abstract class EditorAdornmentWaiter : AsynchronousOperationListener
-    {
-        public override Task CreateWaitTask()
```

Comment:
I'm horrified this was virtual and had a special case like this. Do we still need this logic? (I hope not...?)

Reply:
the way it does was wrong I believe. since it basically wanted to call DoEvent for other waiter (not this waiter). now I explictly call Doevent while waiting.